core/src/test/java/org/elasticsearch/search/aggregations/InternalAggregationTestCase.java

```diff
@@ -203,15 +203,8 @@ public final void testFromXContent() throws IOException {
         assumeTrue("This test does not support the aggregation type yet",
                 getNamedXContents().stream().filter(entry -> entry.name.match(aggregation.getType())).count() > 0);
 
-        final ToXContent.Params params = new ToXContent.MapParams(singletonMap(RestSearchAction.TYPED_KEYS_PARAM, "true"));
-        final boolean humanReadable = randomBoolean();
         final XContentType xContentType = randomFrom(XContentType.values());
-        final BytesReference originalBytes = toShuffledXContent(aggregation, xContentType, params, humanReadable);
-
-        final Aggregation parsedAggregation = parse(aggregation, xContentType, humanReadable, randomBoolean());
-
-        final BytesReference parsedBytes = toXContent((ToXContent) parsedAggregation, xContentType, params, humanReadable);
-        assertToXContentEquivalent(originalBytes, parsedBytes, xContentType);
+        final Aggregation parsedAggregation = parseAndAssert(aggregation, xContentType, randomBoolean(), randomBoolean());
```

Comment:
Nit: could this by simplified even more by randomizing the xContentType, humanReadable and shuffle flag inside of `parsedAggregation`? Not sure about other callers of this method...